src/libs/httpclient/creporternwsessionmgr.cpp

```diff
@@ -122,7 +139,8 @@ bool CReporterNwSessionMgr::canUseNetworkConnection()
 
     return (config.bearerType() == QNetworkConfiguration::BearerWLAN) ||
            (config.bearerType() == QNetworkConfiguration::BearerEthernet) ||
-           !CReporterNwSessionMgrPrivate::connectionIsActive();
+           (!CReporterNwSessionMgrPrivate::connectionIsActive() &&
+            CReporterNwSessionMgrPrivate::developerModeIsActive());
```

Comment:
I'm not sure this developer mode stuff is worthwhile here. It's quite a guesswork anyway. I'd assume very low percentage of devices with developer mode on have really usb networking.

Reply:
The point is that it does not blindly try uploading whenever no network connection is available, producing error events unnecessarily.